cpu/cc3200/cpu.c

```diff
@@ -55,3 +66,110 @@ void reset_periph_clk(cc3200_arcm_reg_t *reg)
     /* deassert reset */
     reg->soft_reset &= ~PRCM_SOFT_RESET;
 }
+
+/**
+ * @brief Get the sys reset cause object
+ *
+ * @return uint32_t restart reason code
+ */
+static uint32_t get_sys_reset_cause(void)
+{
+    uint32_t reset_cause;
+    uint32_t hiber_status;
+    /* reset read reset status */
+    reset_cause = GPRCM->APPS_RESET_CAUSE & 0xFF;
+
+    if (reset_cause == PRCM_POWER_ON) {
+        hiber_status = HWREG(HIB3P3_BASE + HIB3P3_O_MEM_HIB_WAKE_STATUS);
+        /* FIXME: wait for some reason test if this is needed*/
+        ROM_UtilsDelay(PRCM_OP_DELAY);
+        if (hiber_status & 0x1) {
+            return PRCM_HIB_EXIT;
+        }
+    }
+    return reset_cause;
+}
+
+/**
+ * @brief periph_reset resets peripherals
+ * based upon PRCMCC3200MCUInit from TIs prcm.c
+ *
+ */
+static void periph_reset(void)
+{
+    cc3200_reg_t tmp;
+
+    /* DIG DCDC LPDS ECO Enable */
+    HWREG(HIB1P2_BASE + HIB1P2_O_ANA_DCDC_PARAMETERS16) |= 0x800000;
+
+    /* enable hibernate ECO */
+    tmp = HWREG(HIB3P3_BASE + HIB3P3_O_MEM_HIB_WAKE_STATUS);
+    ROM_UtilsDelay(PRCM_OP_DELAY);
+    HWREG(HIB3P3_BASE + HIB3P3_O_MEM_HIB_REG0) = tmp | (1 << 4);
+    ROM_UtilsDelay(PRCM_OP_DELAY);
+
+    /* enable clock switching */
+    HWREG(OCP_SHARED_BASE + OCP_SHARED_O_SPARE_REG_5) |= 0x3C;
+
+    /* enable and reset default peripheral */
+    reset_periph_clk(&ARCM->UDMA_A);
+
+    /* disable udma */
+    ARCM->UDMA_A.clk_gating &= ~PRCM_RUN_MODE_CLK;
+
+    if (get_sys_reset_cause() == PRCM_POWER_ON) {
+        /* enable RTC timer after hibernation */
+        HWREG(HIB3P3_BASE + HIB3P3_O_MEM_HIB_RTC_TIMER_ENABLE) = 0x1;
+        ROM_UtilsDelay(PRCM_OP_DELAY);
+    }
+
+    /* SWD mode */
+    if (((HWREG(HIB1P2_BASE + HIB1P2_O_SOP_SENSE_VALUE) & 0xFF) == 0x2)) {
+        HWREG(OCP_SHARED_BASE + OCP_SHARED_O_GPIO_PAD_CONFIG_28) =
+                ((HWREG(OCP_SHARED_BASE + OCP_SHARED_O_GPIO_PAD_CONFIG_28) &
+                  ~0xC0F) |
+                 0x2);
+        HWREG(OCP_SHARED_BASE + OCP_SHARED_O_GPIO_PAD_CONFIG_29) =
+                ((HWREG(OCP_SHARED_BASE + OCP_SHARED_O_GPIO_PAD_CONFIG_29) &
+                  ~0xC0F) |
+                 0x2);
+    }
+
+    /* Override JTAG mux */
+    HWREG(OCP_SHARED_BASE + OCP_SHARED_O_CC3XX_DEV_PADCONF) |= 0x2;
+
+    /* Change UART pins(55,57) mode to PIN_MODE_0 if they are in PIN_MODE_1
```

Comment:
What is the meaning of PIN_MODE_0  and PIN_MODE_1?
Is this always necessary or should it be done in the UAR driver?

Reply:
PIN_MODE_0 is default/GPIO PIN_MODE_1 is UART. I tried starting the board without that init and it still worked fine... I don't know why it is done that way by TI, but I could simply remove it. The UART driver takes care of it already.